analyses/molecular-subtyping-embryonal/00-embryonal-select-pathology-dx.Rmd

```diff
@@ -0,0 +1,197 @@
+```
+
+We're going to save the pathology diagnosis information we'll use to generate the subset files in a directory `hgg-subset`.
+
+```{r}
+output_dir <- "subset-files"
+output_file <- file.path(output_dir, 
+                         "embryonal_subtyping_path_dx_strings.json")
+```
+
+## Read in data
+
+```{r}
+histologies_df <- read_tsv(histologies_file)
+```
+
+## Explore the pathology diagnoses
+
+### `pathology_diagnosis == Supratentorial or Spinal Cord PNET`
+
+We want to filter rows on the basis of the presence of the values "Supratentorial or Spinal Cord PNET" and "Neuroblastoma" in the `pathology_diagnosis` column, [per this comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/752#issuecomment-697000066).  
+
+```{r}
+histologies_df %>%
+  filter(
+    str_detect(str_to_lower(pathology_diagnosis), "neuroblastoma") |
+      pathology_diagnosis == "Supratentorial or Spinal Cord PNET"
+  ) %>%
+  count(pathology_diagnosis) %>%
+  arrange(desc(n))
+```
+
+For the most part, this is as we would expect given the 2016 WHO classifications.
+When filtering, we will want to keep CNS neuroblastoma samples and not peripheral neuroblastoma samples for this module.
+
+### "Other" samples that should be included based on `pathology_free_text_diagnosis`
+
+We will need to filter on the basis of the `pathology_free_text_diagnosis` column when `pathology_diagnosis == "Other"` and in order to distinguish CNS from peripheral neuroblastomas.
+
+Let's take a look at the free text field under the above conditions.
+
+```{r}
+histologies_df %>% 
+  filter(
+    str_detect(str_to_lower(pathology_diagnosis), "neuroblastoma") |
+      pathology_diagnosis == "Other"
+  ) %>%
+  group_by(pathology_free_text_diagnosis) %>%
+  tally() %>%
+  arrange(desc(n))
+```
+
+We will need to select free text terms for inclusions because above we can say values such as `peripheral neuroblastoma` that we do not want to include in this module.
+
+## Pathology diagnosis strings for inclusion
+
+These are the terms that we'll use to look for exact matches in `pathology_diagnosis`.
+ 
+```{r}
+path_dx_terms <- c(
+  "Supratentorial or Spinal Cord PNET"
+) # "neuroblastoma" will also need to be detected in the `pathology_diagnosis` column, but the logic around it is a bit more complex so we will handle this term separately
+```
+
+## Pathology free text diagnosis strings for inclusion
+
+These are the terms that we'll use to look for exact matches in the `pathology_free_text_diagnosis` column.
+
+```{r}
+free_text_dx_terms <- c(
+  "embryonal tumor with multilayer rosettes, ros (who grade iv)",
+  "embryonal tumor, nos, congenital type",
+  "ependymoblastoma",
+  "medulloepithelioma",
+  "medullooepithelioma" # It was noted that this misspelling was included in the `pathology_free_text_diagnosis` column of the histologies file so we will add it to the JSON file [per this comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/788#issuecomment-700717514)
```

Comment:
@jaclyn-taroni do we want to be strict here to include only what we see now and update as new terms come along or be pre-emptive and also add general terms such as embryonal tumor, PNET, primitive neuroectodermal tumor?

Reply:
I think update as new terms come along - that method seems more _intentional_ to me than adding terms we don't see yet.